src/main/java/org/elasticsearch/index/shard/IndexShardModule.java

```diff
@@ -47,6 +49,9 @@ public IndexShardModule(ShardId shardId, boolean primary, Settings settings) {
         this.settings = settings;
         this.shardId = shardId;
         this.primary = primary;
+        if (settings.get("index.translog.type") != null) {
```

Comment:
it's just a bwc protection, to be nice.. it shoudn't be used so I kept it 'hidden'. can do if you want.